### ISSUE TITLE
Error when using a generic `TypeGuard`

### ISSUE THREAD

**Issue by merisbahti** [microsoft/pyright#3091]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
I'm trying to create a generic function `validate_type` which basically returns a generic `TypeGuard[T]`.

**To Reproduce**
All I had to do is download the latest `pyright` and run the below code through pyright.

**Expected behavior**
I'd expect no error, as vscode pyright plugin has no problems figuring out it is a string (it reports the same error).

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import TypeVar

from typing_extensions import TypeGuard

T = TypeVar("T")


def validate_type(input: object, type: T) -> TypeGuard[T]:
    return True


def f(input: object) -> None:
    if validate_type(input, str):
        print(input.capitalize()) # error: Argument missing for parameter "self" (reportGeneralTypeIssues)
 ```

While the python vscode extension seems to have no problems figuring out the type:
<img width="450" alt="image" src="https://user-images.githubusercontent.com/2946728/154967464-90417f98-b2fa-467d-9b52-88c7388b64cd.png">

**VS Code extension or command-line**
Pyright version: `pyright 1.1.223`

**Additional context**
None

**Comment by thomkeh** [comment]
I think you have an error in the signature of `validate_type`. I should be

```python
def validate_type(input: object, type: Type[T]) -> TypeGuard[T]:
```

**Comment by erictraut** [comment]
Yes, @thomkeh has identified the problem. If you make that change, it will work as expected.

**Comment by merisbahti** [comment]
Wow thanks for that amazingly fast reply.

Still seems that mypy has a problem with this 👎 but that isn't your problem!
<img width="415" alt="image" src="https://user-images.githubusercontent.com/2946728/154983176-f6608f90-1bdb-482d-945f-d5e1921ae89c.png">
